openfe/setup/methods/rbfe_methods.py

```diff
@@ -0,0 +1,257 @@
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    # mapping of component name to forcefield path
+    forcefield: Dict[str, str]
+    nonbonded_electostatics = 'PME'
+    nonbonded_cutoff = 0.9 * unit.nanometer
+    constraints = 'HBonds'
+    rigid_water = True
+
+
+class StatePoint(BaseModel):
+    """Description of the temperature and pressure to model"""
+    class Config:
+        arbitrary_types_allowed = True
+
+    temperature = 298.15 * unit.kelvin
+    pressure = 1 * unit.bar
+
+    @validator('pressure', 'temperature')
+    def must_be_positive(cls, v):
+        if v <= 0:
+            raise ValueError("Pressure and temperature must be positive")
+        return v
+
+    @validator('pressure')
+    def is_pressure(cls, v):
+        if not v.is_compatible_with(unit.bar):
+            raise ValueError("Must be pressure value, e.g. use unit.bar")
+        return v
+
+    @validator('temperature')
+    def is_temperature(cls, v):
+        if not v.is_compatible_with(unit.kelvin):
+            raise ValueError("Must be temperature value, e.g. use unit.kelvin")
+
+
+class LambdaProtocolSettings(BaseModel):
+    """Settings for the lambda protocol
+
+    This describes a fixed number of windows
+
+    Attributes
+    ----------
+    functions : str, default 'default'
+      key of which switching functions to use for alchemical mutation
+    windows : int, default 11
+      number of lambda windows to calculate
+    sample_endstate : bool, default False
+      whether to sample the endstates ???
+    """
+    functions = 'default'
+    windows = 11
+    sample_endstates = False
+
+
+class MonteCarloBarostatSettings(BaseModel):
+    """Settings for the OpenMM Monte-Carlo Barostat
+
+    The temperature and pressure value is taken from the StatePoint variable
+
+    Attributes
+    ----------
+    frequency : unit.timestep
+      the number of timesteps between attempts of changing the pressure
+    """
+    class Config:
+        extra = 'forbid'
+
+    frequency = 50 * unit.timestep
+
+
+class MCMCLangevinSplittingDynamicsMoveSettings(BaseModel):
+    """Settings for the integrator
+
+    Attributes
+    ----------
+    timestep
+      size of timestep
+    collision_rate
+
+    n_steps
+
+    reassign_velocities
+
+    n_restart_attempts
+
+    constraint_tolerance
+    """
+    class Config:
+        arbitrary_types_allowed = True
+
+    timestep = 0.02 * unit.femtosecond
+    collision_rate = 1 / unit.picosecond
+    n_steps = 2500 * unit.timestep
+    reassign_velocities = True
+    n_restart_attempts = 20
+    constraint_tolerance = 1e-06
+
+
+class HybridTopologyFactorySettings(BaseModel):
+    use_dispersion_correction = False
+    softcore_alpha = 0.5
+    softcore_LJ_v2 = True
+    softcore_LJ_v2_alpha = 0.85
+    softcore_electrostatics = True
+    softcore_electrostatics_alpha = 0.3
+    softcore_sigma_Q = 1.0
+    interpolate_old_and_new_14s = False
+    flatten_torsions = False
```

Comment:
I would be tempted to say here "leave it for now with an expected refactor soon enough". The likelihood is the way this is all handled is going to change drastically once we come up with a reasonable way to put all this into openmmtools. I know it's be nice to fix now, and if we have the time we should, but whatever we do now is based on a class that is going to die soon no matter what.

Reply:
I think `softcore_LJ_v2` is that use v2 setting, no? That works (can remove duplication of `softcore_alpha`), but I think the better long-term solution is a separate object. What happens when someone comes up with v3? Better to have it be modular and replaceable.

Reply:
I think long term folks will be passing their own defined functions for these (at least that's what openmmtools does already, so if we go another route then that'll be up to to use to convince them otherwise).

But yeah, a "pseudo-reasonable short term solution that works for now" is the best option here imho.